packages/block-library/src/categories/block.json

```diff
@@ -2,11 +2,15 @@
 	"$schema": "https://schemas.wp.org/trunk/block.json",
 	"apiVersion": 3,
 	"name": "core/categories",
-	"title": "Categories List",
+	"title": "Terms List",
```

Comment:
Given that we are changing the name of the block here I would recommend we at least add `Categories` to the `keywords` array so it still appears in search. 

---

My personal recommendation however would be that for backwards compatibility we register one singular block variation for the category taxonomy. This way we are not renaming an existing block. Users still get the same result. But they now have an additional control to change the block to query other taxonomies.

Reply:
> Given that we are changing the name of the block here I would recommend we at least add `Categories` to the `keywords` array so it still appears in search.

Good point 👍 

> My personal recommendation however would be that for backwards compatibility we register one singular block variation for the category taxonomy. This way we are not renaming an existing block. Users still get the same result. But they now have an additional control to change the block to query other taxonomies.

Interesting. I gave this a quick spin (see below diff). I'm assuming you mean with `scope` set to exclude `transform`, so that there won't be the dropdown to transform variations (as there is in #64805) as it'd be otherwise confusing to have two dropdowns to control which variation to display.

This posed the following problem: For back-compat, we need to keep `category` as the default value for the `taxonomy` attribute, so that any existing occurrences of `<!-- wp:categories /-->` will continue to be rendered as the Categories List variation.

But that raises the question of the default value for that attribute when inserting a new instance of the generic Terms List variation. It cannot be omitted, as that would translate to `category` (per the above), causing the block to be promptly rendered as the Categories List variation.

So we need a different default value for the Terms List variation. I guess Tags -- the only other built-in and public taxonomy -- is the logical contender here, so this is what I chose.

Furthermore, there's the question of the dropdown. IMO, it needs to be hidden altogether in case of the Categories List variation, as it would otherwise allow to transform it to the Terms List variation. In the same vein, the "Categories" option needs to be removed from the dropdown in case of the Terms List variation, to prevent transforming into the Categories List variation.

But that means that the Terms List block is now solely for Tags and custom taxonomies, whereas Categories get their own variation (which looks like a separate block, for all intents and purposed). This feels a bit arbitrary IMO; I'm not sure it's that much better in terms of UX than _not_ having the separate Categories List block variation.

<details>
<summary>
Diff
</summary>

```diff
diff --git a/packages/block-library/src/categories/edit.js b/packages/block-library/src/categories/edit.js
index 866feb884c8..bafb2a3273a 100644
--- a/packages/block-library/src/categories/edit.js
+++ b/packages/block-library/src/categories/edit.js
@@ -46,7 +46,11 @@ export default function CategoriesEdit( {
 		'taxonomy'
 	);
 
-	const taxonomies = allTaxonomies?.filter( ( t ) => t.visibility.public );
+	const taxonomies = allTaxonomies?.filter(
+		( t ) =>
+			t.visibility.public &&
+			( taxonomySlug === 'category' ) === ( t.slug === 'category' )
+	);
 
 	const taxonomy = taxonomies?.find( ( t ) => t.slug === taxonomySlug );
 
@@ -185,21 +189,24 @@ export default function CategoriesEdit( {
 		<TagName { ...blockProps }>
 			<InspectorControls>
 				<PanelBody title={ __( 'Settings' ) }>
-					{ Array.isArray( taxonomies ) && (
-						<SelectControl
-							__nextHasNoMarginBottom
-							__next40pxDefaultSize
-							label={ __( 'Taxonomy' ) }
-							options={ taxonomies.map( ( t ) => ( {
-								label: t.name,
-								value: t.slug,
-							} ) ) }
-							value={ taxonomySlug }
-							onChange={ ( selectedTaxonomy ) =>
-								setAttributes( { taxonomy: selectedTaxonomy } )
-							}
-						/>
-					) }
+					{ taxonomySlug !== 'category' && // For back-compat, the Category taxonomy has its own block variation.
+						Array.isArray( taxonomies ) && (
+							<SelectControl
+								__nextHasNoMarginBottom
+								__next40pxDefaultSize
+								label={ __( 'Taxonomy' ) }
+								options={ taxonomies.map( ( t ) => ( {
+									label: t.name,
+									value: t.slug,
+								} ) ) }
+								value={ taxonomySlug }
+								onChange={ ( selectedTaxonomy ) =>
+									setAttributes( {
+										taxonomy: selectedTaxonomy,
+									} )
+								}
+							/>
+						) }
 					<ToggleControl
 						__nextHasNoMarginBottom
 						label={ __( 'Display as dropdown' ) }
diff --git a/packages/block-library/src/categories/index.js b/packages/block-library/src/categories/index.js
index 8cdcad45086..d30c55667d1 100644
--- a/packages/block-library/src/categories/index.js
+++ b/packages/block-library/src/categories/index.js
@@ -9,6 +9,7 @@ import { category as icon } from '@wordpress/icons';
 import initBlock from '../utils/init-block';
 import metadata from './block.json';
 import edit from './edit';
+import variations from './variations';
 
 const { name } = metadata;
 
@@ -18,6 +19,7 @@ export const settings = {
 	icon,
 	example: {},
 	edit,
+	variations,
 };
 
 export const init = () => initBlock( { name, metadata, settings } );
diff --git a/packages/block-library/src/categories/variations.js b/packages/block-library/src/categories/variations.js
new file mode 100644
index 00000000000..06003ec6420
--- /dev/null
+++ b/packages/block-library/src/categories/variations.js
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { category as icon } from '@wordpress/icons';
+
+const variations = [
+	{
+		name: 'terms',
+		title: __( 'Terms List' ),
+		icon,
+		attributes: {
+			// We need to set an attribute here that will be set when inserting the block.
+			// We cannot leave this empty, as that would be interpreted as the default value,
+			// which is `category` -- for which we're defining a distinct variation below,
+			// for backwards compatibility reasons.
+			// The logical fallback is thus the only other built-in and public taxonomy: Tags.
+			taxonomy: 'post_tag',
+		},
+		isActive: ( blockAttributes ) =>
+			// This variation is used for any taxonomy other than `category`.
+			blockAttributes.taxonomy !== 'category',
+	},
+	{
+		name: 'categories',
+		title: __( 'Categories List' ),
+		description: __( 'Display a list of all categories.' ),
+		icon,
+		attributes: {
+			taxonomy: 'category',
+		},
+		isActive: [ 'taxonomy' ],
+	},
+];
+
+export default variations;

```
</details>

![separate-categories-variation](https://github.com/user-attachments/assets/5c096cbe-c4af-49f2-8ecf-d7ac1ee64a6d)

Reply:
Looking at the diff I would not remove the `category` option from the dropdown of available taxonomies. If a user happens to insert the "Term List" block and then selects the "Category" taxonomy in the drop-down I'd be fine with that changing the name of the block to the correct variation.

Reply:
At the same time I would be fine with users switching from the Category List to another term :)

Reply:
TBH, that would feel inconsistent to me in terms of UX; I think it would be somewhat surprising to the user if selecting "Categories" as the taxonomy from that dropdown gets special treatment, while all other taxonomies don't 😕 

![dropdown-includes everything](https://github.com/user-attachments/assets/69b61f8d-3222-46bf-9c61-e212617108ad)

If it's not too much of a dealbreaker for you, I'd rather consider the idea of a dedicated variation for the Categories List block separately -- i.e. in a follow-up PR, where it can also get the attention by designers that it deserves, without conflating it with the principal subject of this PR of adding the `taxonomy` attribute and dropdown in the first place 😊

Reply:
Works for me 👍

Reply:
Thank you!

I've added `"keywords": [ "categories" ]` in https://github.com/WordPress/gutenberg/pull/65272/commits/95a6cf35fc4c01549c627b06f003404818346c7f.

Reply:
It looks like block variations were discussed extensively in:
- https://github.com/WordPress/gutenberg/pull/64805

One drawback is that users won't see Categories in the inserters when typing `categories` term. The icon remains the same, so maybe that isn't that much of an issue, as with the `keywords`, it will remain discoverable. Although, we should follow the feedback closely during beta and RC phase of the release.

Reply:
That's a fair point and one that I hadn't considered; I've shared some [more thoughts](https://github.com/WordPress/gutenberg/pull/65272#issuecomment-2356367879) below.

I'll add one more note for posterity: [With the suggested change](https://github.com/WordPress/gutenberg/pull/65272#discussion_r1761135864) we've been discussing in this thread -- i.e. adding _two_ block variations, one for Categories specifically, and a "generic" one for all other cases -- there can also be some confusion in the inserter: I've called the "generic" variation "Terms List", just like the block itself; but since both the block itself and all its variations are shown in the inserter, there would be _two_ "Terms List" blocks:

![image](https://github.com/user-attachments/assets/4baaa929-e849-40fe-8fe2-2ac5cc71d80d)

Reply:
I've filed a draft PR for this regardless, in case we'd like to revisit this: https://github.com/WordPress/gutenberg/pull/65434